src/client/eppo-client.spec.ts

```diff
@@ -596,13 +600,223 @@ describe('EppoClient E2E test', () => {
 
     const client = new EppoClient(storage);
     let assignment = client.getAssignment('subject-10', flagKey, { appVersion: 9 });
-    expect(assignment).toEqual(null);
+    expect(assignment).toBeNull();
     assignment = client.getAssignment('subject-10', flagKey);
-    expect(assignment).toEqual(null);
+    expect(assignment).toBeNull();
     assignment = client.getAssignment('subject-10', flagKey, { appVersion: 11 });
     expect(assignment).toEqual('control');
   });
 
+  it('returns control variation and logs status_quo if subject is in holdout', () => {
+    const entry = {
+      ...mockExperimentConfig,
+      allocations: {
+        allocation1: {
+          percentExposure: 1,
+          holdout: {
+            statusQuo: 'variation-7',
+            shipped: null,
+            percentExposure: 0.2,
+            keys: [
+              {
+                holdoutKey: 'holdout-2',
+                statusQuoShardRange: {
+                  start: 0,
+                  end: 50,
+                },
+              },
+              {
+                holdoutKey: 'holdout-3',
+                statusQuoShardRange: {
+                  start: 51,
+                  end: 100,
```

Comment:
I may be misunderstanding things, but I think all we really need is percent exposure for the holdout. 

If we are trying to save that step in the SDK, I think each shard range then reflect the percentage. So if we have two overlapping holdouts at 2% and 5%, they'd have total ranges 0 - 200 and 0 - 500, with shipped and status quo being subdivisions in there.

The way you have it now makes it seem like a user could only be eligible for one holdout, which I don't think makes sense. Weird things could happen when the earlier holdout ends and now it's just down to one.

Reply:
I think the change that would have to happen with only using percent exposures without any shard subdivision is that a single subject could have multiple holdouts applied to it and then we'd send a list of holdout assignments and write multiple rows in the assignments table. Is that what you're thinking?